### ISSUE TITLE
Google Sign-In JavaScript platform library deprecation

### ISSUE THREAD

**Issue by Brunom50** [FusionAuth/fusionauth-issues#1939]
---
Name: Google Sign-In JavaScript platform library deprecation
About: According to Google the Google Sign-In JavaScript platform library will be deprecated in March 31, 2023.

---

Hello.

According to Google, the Google Sign-In JavaScript platform library will be deprecated in March 31, 2023. Will this affect the Google SSO in FusionAuth? Do we need to update to a later version or does Fusion Auth use another library to handle the google login?

Thanks.

### Related
- https://github.com/FusionAuth/fusionauth-issues/issues/1894
- https://github.com/FusionAuth/fusionauth-issues/issues/1911

### Documentation
- [ ] https://github.com/FusionAuth/fusionauth-site/issues/1977

**Comment by mooreds** [comment]
Hi @Brunom50,

Thanks for using FusionAuth!

Does https://github.com/FusionAuth/fusionauth-issues/issues/1894 address your concerns?

If so, please feel free to close this one. If not, please feel free to share how they are different.

**Comment by harishreddy-m** [comment]
The deadline is very close. Do you have any solution? (Even if it involves manually updating theme)

**Comment by forteilgmbh** [comment]
`<script src="https://apis.google.com/js/api:client.js"></script>` 
in Theme>>helpers should be replaced with 
  `<script src="https://accounts.google.com/gsi/client" async defer></script>`
  
I have made this change. 

However, source code in `/js/identityProvider/Google.js?version=1.30.1`

```
gapi.load('auth2', function() {
      // Retrieve the singleton for the GoogleAuth library and set up the client.
      var auth2 = gapi.auth2.init({
        client_id: this.clientId,
        scope: this.scope
      });

      auth2.attachClickHandler(this.button.domElement, {}, this._onSignIn, this._onFailure);
    }.bind(this));

```

should be replaced with 
```
google.accounts.id.initialize({
      client_id: 'YOUR_GOOGLE_CLIENT_ID',
      callback: handleCredentialResponse
    });
    google.accounts.id.prompt();
    ```
    This is where we need your help.

**Comment by robotdan** [comment]
Thanks @forteilgmbh ! We'll be fixing this under https://github.com/FusionAuth/fusionauth-issues/issues/1894.

**Comment by spwitt** [comment]
**UPDATE**

FusionAuth offers two login methods for the Google IdP: redirect and popup. These can be configured in _Settings > Identity Providers > Edit Google_.

If your Google IdP is configured to _Use redirect for login_, then **you are not impacted by the upcoming library deprecation**.

If your Google IdP is configured to _Use popup for login_, then prior to the March 31, 2023 deadline you must either:

1. Change your login method to _Use redirect for login_
2. Update to a new version of FusionAuth with the Google IdP client-side fixes (in progress)

![image](https://user-images.githubusercontent.com/3409780/220948851-bcb0513b-41db-4302-9bf7-ca7d0c533222.png)

_More info_

I used Google's `G_AUTH2_MIGRATION` cookie with a value of `informational` ([link](https://developers.google.com/identity/gsi/web/guides/migration#identifying_affected_code_and_testing)) to log console warnings when it detects that the client invokes deprecated JavaScript APIs. Based on this testing I found that:

* When the login method is _Use redirect for login_, nothing is logged to the console
* When the login method is _Use popup for login_, there is a message logged to the console

```
Your client application uses libraries for user authentication or authorization that will soon be deprecated. See the [Migration Guide](https://developers.google.com/identity/gsi/web/guides/gis-migration) for more information.
```

I tested and observed the same behavior across the following browsers on macOS:
* Chrome
* Firefox
* Safari
* Brave
* Edge

**Comment by spwitt** [comment]
Internal
- https://github.com/FusionAuth/fusionauth-app/pull/216

**Comment by mooreds** [comment]
Related forum post: https://fusionauth.io/community/forum/topic/2329/upcoming-google-identity-provider-changes

**Comment by spwitt** [comment]
Feature released in 1.44.0. Documentation changes have been merged. Closing the issue.
